<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>IMUCostumeConfigurationWizard</class>
 <widget class="QWizard" name="IMUCostumeConfigurationWizard">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>624</width>
    <height>732</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Costume Calibration Wizard</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../../imu_view/ui/MdeViewIcons.qrc">
    <normaloff>:/mde/icons/imu.png</normaloff>:/mde/icons/imu.png</iconset>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <property name="wizardStyle">
   <enum>QWizard::AeroStyle</enum>
  </property>
  <property name="options">
   <set>QWizard::HelpButtonOnRight|QWizard::NoBackButtonOnStartPage</set>
  </property>
  <widget class="QWizardPage" name="costumeInfoPage">
   <property name="title">
    <string>Costume info</string>
   </property>
   <property name="subTitle">
    <string/>
   </property>
   <layout class="QFormLayout" name="formLayout">
    <property name="fieldGrowthPolicy">
     <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
    </property>
    <item row="0" column="0">
     <widget class="QLabel" name="ipAddressLabel">
      <property name="text">
       <string>IP Address:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="imuSensorsLabel">
      <property name="text">
       <string>IMU Sensors:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="grfSensorsLabel">
      <property name="text">
       <string>GRF Sensors:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="imuSensorsCountPlaceholderLabel">
      <property name="text">
       <string>Count</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLabel" name="grfSensorsCountPlaceholderLabel">
      <property name="text">
       <string>Count</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="ipPlaceholderLabel">
      <property name="text">
       <string>IP</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="modelConfigurationPage">
   <property name="title">
    <string>Model configuration page</string>
   </property>
   <layout class="QFormLayout" name="formLayout_4">
    <item row="0" column="0">
     <widget class="QLabel" name="modelClassLabel">
      <property name="text">
       <string>Class:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QComboBox" name="modelClassComboBox"/>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="modelLabel">
      <property name="text">
       <string>Model:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QComboBox" name="modelComboBox"/>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="createModelPushButton">
      <property name="text">
       <string>Create</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <spacer name="horizontalSpacer_5">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>504</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="modelPhisicalDimensionsPage">
   <property name="minimumSize">
    <size>
     <width>525</width>
     <height>510</height>
    </size>
   </property>
   <property name="title">
    <string>Model physical dimensions</string>
   </property>
   <property name="subTitle">
    <string/>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="bodyHeighLabel">
          <property name="text">
           <string>Body Height:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="bodyHeightDoubleSpinBox">
          <property name="toolTip">
           <string>Ground to top of head when standing upright</string>
          </property>
          <property name="value">
           <double>1.850000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="footSizeLabel">
          <property name="text">
           <string>Foot Size:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="footSizeDoubleSpinBox">
          <property name="toolTip">
           <string>Top of shoe nose to end of the heel</string>
          </property>
          <property name="value">
           <double>0.260000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="ankleHeightLabel">
          <property name="text">
           <string>Ankle Height:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="ankleHeightDoubleSpinBox">
          <property name="toolTip">
           <string>Ground to distal tip of lateral malleolus</string>
          </property>
          <property name="value">
           <double>0.120000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="hipHeightLabel">
          <property name="text">
           <string>Hip Height:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="hipHeightDoubleSpinBox">
          <property name="toolTip">
           <string>Ground to most lateral bony prominence of greater trochanter</string>
          </property>
          <property name="value">
           <double>0.890000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="hipWidthLabel">
          <property name="text">
           <string>Hip Width:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QDoubleSpinBox" name="hipWidthDoubleSpinBox">
          <property name="toolTip">
           <string>Right to left anterior sup. iliac spine</string>
          </property>
          <property name="value">
           <double>0.340000000000000</double>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="kneeHeightLabel">
          <property name="text">
           <string>Knee Height:</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QDoubleSpinBox" name="kneeHeightDoubleSpinBox">
          <property name="toolTip">
           <string>Ground to lateral epicondyle on the femoral bone</string>
          </property>
          <property name="value">
           <double>0.320000000000000</double>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="shoulderWidthLabel">
          <property name="text">
           <string>Shoulder Width:</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QDoubleSpinBox" name="shoulderWidthDoubleSpinBox">
          <property name="toolTip">
           <string>Right to left distal tip of acromion (acromial angle)</string>
          </property>
          <property name="value">
           <double>0.430000000000000</double>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="shoeSoleHeightLabel">
          <property name="text">
           <string>Shoe Sole Height:</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QDoubleSpinBox" name="shoeSoleHeightDoubleSpinBox">
          <property name="toolTip">
           <string>Average thickness of the sole of the used shoes. This value will add an offset to the ankle height.</string>
          </property>
          <property name="value">
           <double>0.040000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="bodyDimensionsLabel">
      <property name="minimumSize">
       <size>
        <width>375</width>
        <height>500</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>375</width>
        <height>500</height>
       </size>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap resource="imuCostumeIcons.qrc">:/imuCostume/icons/skeletonWithDimensions.png</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::NoTextInteraction</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="sensorsMappingPage">
   <property name="title">
    <string>Sensors to model mapping</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QTableWidget" name="sensorsMappingTableWidget">
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Sensor</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Joint</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Offset</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Orientation</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="oneToOneMappingCheckBox">
      <property name="text">
       <string>One-To-One mapping</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="sensorsEstimationPage">
   <property name="title">
    <string>Sensors` orientation estimation</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QWidget" name="singleEstimationWidget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="singleEstimationLabel">
         <property name="text">
          <string>Orientation estimation algorithm:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="singleEstimationAlgorithmComboBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTableWidget" name="sensorsEstimationAlgorithmsTableWidget">
      <column>
       <property name="text">
        <string>Sensor</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Algorithm</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="singleEstimationAlgorithmCheckBox">
      <property name="text">
       <string>Single estimation algorithm</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="calibrationPage">
   <property name="title">
    <string>Calibration</string>
   </property>
   <layout class="QFormLayout" name="formLayout_5">
    <item row="0" column="0">
     <widget class="QLabel" name="calibrationAlgorithLabel">
      <property name="text">
       <string>Calibration algorithm:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QComboBox" name="calibrationAlgorithmComboBox"/>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="motionEstimationPage">
   <property name="title">
    <string>Motion estimation</string>
   </property>
   <layout class="QFormLayout" name="formLayout_3">
    <item row="0" column="0">
     <widget class="QLabel" name="motionEstimationAglorithmLabel">
      <property name="text">
       <string>Motion estimation algorithm:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QComboBox" name="motionEstimationAglorithmComboBox"/>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="../../../imu_view/ui/MdeViewIcons.qrc"/>
  <include location="imuCostumeIcons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>singleEstimationAlgorithmCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>IMUCostumeConfigurationWizard</receiver>
   <slot>onEstimationAlgorithmSelectionModeChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>IMUCostumeConfigurationWizard</sender>
   <signal>currentIdChanged(int)</signal>
   <receiver>IMUCostumeConfigurationWizard</receiver>
   <slot>pageChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sensorsMappingTableWidget</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>IMUCostumeConfigurationWizard</receiver>
   <slot>verifySensorsMapping(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>387</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sensorsEstimationAlgorithmsTableWidget</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>IMUCostumeConfigurationWizard</receiver>
   <slot>verifyEstimationAlgorithms(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modelComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>IMUCostumeConfigurationWizard</receiver>
   <slot>verifyModel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>calibrationAlgorithmComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>IMUCostumeConfigurationWizard</receiver>
   <slot>verifyCalibrationAlgorithm(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>374</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>motionEstimationAglorithmComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>IMUCostumeConfigurationWizard</receiver>
   <slot>verifyMotionEstimationAlgorithm(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onEstimationAlgorithmSelectionModeChange(int)</slot>
  <slot>onSensorsMappingCellDoubleClick(int,int)</slot>
  <slot>pageChanged(int)</slot>
  <slot>verifyModel(int)</slot>
  <slot>verifySensorsMapping(int,int)</slot>
  <slot>verifyEstimationAlgorithms(int,int)</slot>
  <slot>verifyCalibrationAlgorithm(int)</slot>
  <slot>verifyMotionEstimationAlgorithm(int)</slot>
 </slots>
</ui>
